src/tilemap_layer.cpp

```diff
@@ -141,35 +142,66 @@ TilemapLayer::TilemapLayer(int ilayer) :
 	substitutions(ilayer >= 1
 			? Main_Data::game_data.map_info.upper_tiles
 			: Main_Data::game_data.map_info.lower_tiles),
-	layer(ilayer) {
-
-	memset(autotiles_ab, 0, sizeof(autotiles_ab));
-	memset(autotiles_d, 0, sizeof(autotiles_d));
-
-	// SubLayer for the tiles with Wall or Above passability
-	// Its z-value should be between the z of the events in the upper layer and the hero
-	sublayers.push_back(std::make_shared<TilemapSubLayer>(this, Priority_TilesetAbove + layer));
+	layer(ilayer),
 	// SubLayer for the tiles without Wall or Above passability
 	// Its z-value should be under z of the events in the lower layer
-	sublayers.push_back(std::make_shared<TilemapSubLayer>(this, Priority_TilesetBelow + layer));
+	lower_layer(this, Priority_TilesetBelow + layer),
+	// SubLayer for the tiles with Wall or Above passability
+	// Its z-value should be between the z of the events in the upper layer and the hero
+	upper_layer(this, Priority_TilesetAbove + layer)
+{
 }
 
-void TilemapLayer::DrawTile(Bitmap& dst, Bitmap& screen, int x, int y, int row, int col, bool allow_fast_blit) {
-	Bitmap::TileOpacity op = screen.GetTileOpacity(row, col);
+EP_ALWAYS_INLINE
+void TilemapLayer::DrawTile(Bitmap& dst, Bitmap& tileset, Bitmap& tone_tileset, int x, int y, int row, int col, uint32_t tone_hash, bool allow_fast_blit) {
+	auto op = tileset.GetTileOpacity(col, row);
```

Comment:
And here it was inverted to Get(col, row)

Reply:
The tile opacity code is now in terms of (x,y) so this has to pass (col, row)